scripts/gha/test_simulator.py

```diff
@@ -76,64 +101,89 @@ def main(argv):
-    config = json.load(config)
-
-  if not config:
-    logging.info("No config found")
-    return 1
-
-  logging.info("Config found: %s", config)
-
-  testapps = []
-  for file_dir, directories, _ in os.walk(testapp_dir):
+  
+  ios_testapps = []
+  android_testapps = []
+  for file_dir, directories, file_names in os.walk(testapp_dir):
     # .app is treated as a directory, not a file in MacOS
     for directory in directories:
       full_path = os.path.join(file_dir, directory)
       if "simulator" in full_path and directory.endswith(".app"):
-        testapps.append((_get_bundle_id(full_path, config), full_path))
+        ios_testapps.append(full_path)
+    for file_name in file_names:
+      full_path = os.path.join(file_dir, file_name)
+      if file_name.endswith(".apk"):
+        android_testapps.append(full_path)    
 
-  if not testapps:
+  if not ios_testapps and not android_testapps:
     logging.info("No testapps found")
     return 1
 
-  logging.info("Testapps found: %s", "\n".join(path for _, path in testapps))
+  tests = []
+  if ios_testapps:
+    logging.info("iOS Testapps found: %s", "\n".join(path for path in ios_testapps))
+    
+    ios_device = FLAGS.ios_device
+    device_info = ios_device.split(";")
+    device_name = device_info[0]
+    device_os = device_info[1]
+
+    device_id = _boot_simulator(device_name, device_os)
+    if not device_id:
+      logging.error("simulator created fail")
+      return 1
+
+    # A tool that enable game-loop test. This is a XCode project
+    ios_gameloop_project = os.path.join(current_dir, "integration_testing", "gameloop")
+    ios_gameloop_app = _build_ios_gameloop(ios_gameloop_project, device_name, device_os)
+    if not ios_gameloop_app:
+      logging.error("gameloop app not found")
+      return 2
+
+    config_path = os.path.join(current_dir, "integration_testing", "build_testapps.json")
+    with open(config_path, "r") as configFile:
+      config = json.load(configFile)
+    if not config:
+      logging.error("No config file found")
+      return 3
   
-  gameloop_app = _build_gameloop(gameloop_project, device_name, device_os)
-  if not gameloop_app:
-    logging.info("gameloop app not found")
-    return 2
+    for app_path in ios_testapps:
+      bundle_id = _get_bundle_id(app_path, config)
+      tests.append(Test(
+                      testapp_path=app_path, 
+                      logs=_run_ios_gameloop_test(bundle_id, app_path, ios_gameloop_app, device_id)))
 
-  device_id = _boot_simulator(device_name, device_os)
-  if not device_id:
-    logging.info("simulator created fail")
-    return 3
+  if android_testapps:
+    logging.info("Android Testapps found: %s", "\n".join(path for path in android_testapps))
 
-  tests = []
-  for bundle_id, app_path in testapps:
-    tests.append(Test(
-                    testapp_path=app_path, 
-                    logs=_run_gameloop_test(bundle_id, app_path, gameloop_app, device_id)))
+    android_device = FLAGS.android_device
```

Comment:
Give an example of what's being split out here?

Reply:
https://github.com/firebase/firebase-cpp-sdk/blob/feature/test-on-android-emulator/scripts/gha/test_simulator.py#L56